Add helpful log output to aid with troulbeshooting of vault login.

damienjoldersma-rakuten opened the pull request:
I had some trouble with setting up Vault and Rundeck integration - I was receiving a NullPointerException in the logs with no other information to help me understand the problem of my misconfiguration.  

This pull request will help others in troulbeshooting misconfiguration or other Vault / Rundeck integration problems.  Basically it changes the plugin to not ignore and not to just swallow login exceptions.

Here are some example of the exception output that were being hidden which were essential in understanding to resolve the misconfiguration: 

```
[2024-03-11T02:07:26,527] WARN  vault-storage [qtp689905739-81] - Error logging into Vault: Encountered error while authenticating with approle: Vault responded with HTTP status code: 400
Response body: {"errors":["invalid role ID"]}
```

```
[2024-03-11T02:12:48,291] WARN  vault-storage [pool-12-thread-1] - Error logging into Vault: Encountered error while authenticating with approle: Vault responded with HTTP status code: 400
Response body: {"errors":["invalid secret id"]}
```

fdevans commented:
@damienjoldersma-rakuten - Thank you for the PR.  If you are able to address the CoPilot feedback we can consider moving this forward in the review process.

fdevans commented:
going to close since we haven't heard back.  Let us know if you want to reopen.

damienjoldersma-rakuten commented:
Hi @fdevans, sorry for my delay! I missed the notification obviously ;) 

I've addressed all of the Copilot feedback:

- Downgraded operational lookup/login log messages from `INFO` to `FINE` — `lookup()` is called on every storage operation so INFO was too noisy
- Moved success log messages inside the `try` blocks so they only fire when no exception occurred
- Removed duplicate `e.printStackTrace()` calls — the logger already captures the full stack trace via `log.log(WARNING, ..., e)`
- Fixed typo: "succesfully" → "successfully"

Here is the commit: https://github.com/rundeck-plugins/vault-storage/commit/f0259ef565ba1d763a77bccb7b2f4c2629b9d90b 

I think if you reopen the PR it will update properly and pick up the commit.  

Would you be able to reopen the PR? Happy to make any further changes if needed.

damienjoldersma-rakuten commented on src/main/java/io/github/valfadeev/rundeck/plugin/vault/VaultStoragePlugin.java:
```diff
@@ -314,30 +315,45 @@ private boolean isDir(String key) {
+        } catch (Exception e) {
+            log.log(Level.WARNING, "Caught Exception: " + e.getMessage(), e);
             e.printStackTrace();
         }
+        log.info("Finished Vault lookup succesfully.");
     }
```
Fixed.

damienjoldersma-rakuten commented on src/main/java/io/github/valfadeev/rundeck/plugin/vault/VaultStoragePlugin.java:
```diff
@@ -314,30 +315,45 @@ private boolean isDir(String key) {
                 loginVault(clientProvider);
             } else {
+                log.log(Level.WARNING, "Not a 403 error: " + e.getMessage(), e);
                 e.printStackTrace();
             }
         } catch (ConfigurationException e) {
+            log.log(Level.WARNING, "Caught ConfigurationException: " + e.getMessage(), e);
+            e.printStackTrace();
+        } catch (Exception e) {
+            log.log(Level.WARNING, "Caught Exception: " + e.getMessage(), e);
             e.printStackTrace();
```
Fixed.

damienjoldersma-rakuten commented on src/main/java/io/github/valfadeev/rundeck/plugin/vault/VaultStoragePlugin.java:
```diff
@@ -314,30 +315,45 @@ private boolean isDir(String key) {
     }
 
     protected void lookup(){
+        log.info("Beginning Vault lookup...");
         try {
+            log.info("Will try to lookup self...");
             LookupResponse lookupSelf = getVaultClient().auth().lookupSelf();
```
Fixed.

damienjoldersma-rakuten commented on src/main/java/io/github/valfadeev/rundeck/plugin/vault/VaultStoragePlugin.java:
```diff
@@ -314,30 +315,45 @@ private boolean isDir(String key) {
+            log.log(Level.WARNING, "Caught Exception: " + e.getMessage(), e);
             e.printStackTrace();
         }
+        log.info("Finished Vault lookup succesfully.");
```
Fixed.

damienjoldersma-rakuten commented on src/main/java/io/github/valfadeev/rundeck/plugin/vault/VaultStoragePlugin.java:
```diff
@@ -314,30 +315,45 @@ private boolean isDir(String key) {
 
     private void loginVault(VaultClientProvider provider){
+        log.info("Logging into Vault...");
         try{
+            log.info("Getting Vault Client from provider");
             vaultClient = provider.getVaultClient();
+            log.info("Calling vaultClient logical");
             vault = vaultClient.logical();
         }
-        catch (Exception ignored){
-
+        catch (Exception e){
+            log.log(Level.WARNING, "Error logging into Vault: " + e.getMessage(), e);
         }
+        log.info("Logged into Vault successfully");
     }
```
Fixed.
